src/main/java/se/sundsvall/supportmanagement/service/AccessControlService.java

```diff
@@ -71,4 +84,38 @@ public Specification<ErrandEntity> withAccessControl(String namespace, String mu
 	private boolean hasAccessControlActive(String namespace, String municipalityId) {
 		return namespaceConfigService.get(namespace, municipalityId).isAccessControl();
 	}
+
+	/**
+	 * Fetches ErrandEntity and checks user access, if enabled in namespace.
+	 *
+	 * @param  namespace              namespace
+	 * @param  municipalityId         municipality id
+	 * @param  id                     errand id
+	 * @param  lock                   db row locking enable if true
+	 * @param  accessLevelEnumsFilter filters errands base on user access level for specific errand. If empty, no filtering
+	 *                                occurs (fetches errand if any level exists)
+	 * @return                        errand entity
+	 */
+	public ErrandEntity getErrand(final String namespace, final String municipalityId, final String id, boolean lock, Access.AccessLevelEnum... accessLevelEnumsFilter) {
```

Comment:
Make util/facade method with the name "verifyExistingErrand" or something like that, for increased readability?